src/coreclr/jit/emit.h

```diff
@@ -779,8 +779,10 @@ class emitter
         unsigned _idCallAddr : 1; // IL indirect calls: can make a direct call to iiaAddr
         unsigned _idNoGC : 1;     // Some helpers don't get recorded in GC tables
 #if defined(TARGET_XARCH)
-        unsigned _idEvexbContext : 1; // does EVEX.b need to be set.
-#endif                                //  TARGET_XARCH
+        unsigned _idIsEmbBroadcast : 1; // does EVEX.b need to be set for embedded broadcast.
+        unsigned _idIsEmbRounding : 1;  // does EVEX.b need to be set for embedded rounding.
```

Comment:
Do we need both of these?

My understanding is that `EVEX.b` can be either embedded broadcast or embedded rounding, but not both. Which it refers to then depends on whether it is `register, register` or `load+op`

Which is to say:
```
vaddps zmm0, zmm0, zmm0, {rd-sae}                    ; valid, use round down (toward -inf)
vaddps zmm0, zmm0, dword ptr [rax] {1to16}           ; valid, use embedded broadcast of 1 to 16 elements

vaddps xmm0, xmm0, xmm0, {rd-sae}                    ; invalid, embedded rounding is only valid for zmm and scalar
vaddps zmm0, zmm0, dword ptr [rax] {1to16}, {rd-sae} ; invalid, cannot combine embedded rounding and embedded broadcast
```

Notably, .NET already doesn't support IEEE 754 exceptions (setting `MXCSR.*M` is undefined behavior) and does not support the rounding mode being globally changed from the default (setting MXCSR.RC is undefined behavior), so we can additionally optimize out `{sae}` and `{rne-sae}` and just emit the regular instruction, so I think we actually only need `1` additional bit

That is, given the restrictions .NET has around MXCSR, we could have the following:
```cpp
if (insDesc->idHasMem())
{
    // 0b00 - no embedded broadcast
    // 0b01 - embedded broadcast
    // 0b10 - invalid
    // 0b11 - invalid
}
else
{
    // 0b00 - no embedded rounding
    // 0b01 - {rd-sae}, invalid for EA_ATTR 16/32 only usable for ZMM and scalar on supported instructions
    // 0b10 - {ru-sae}, invalid for EA_ATTR 16/32 only usable for ZMM and scalar on supported instructions
    // 0b11 - {rz-sae}, invalid for EA_ATTR 16/32 only usable for ZMM and scalar on supported instructions
}
```

I expect at least one of the other bits might be "unused" for SIMD register to register scenarios, so we probably don't even need to reduce the smallCns space here. We would however have to identify and ensure it is actually unused and won't cause any other code paths to be accidentally taken.